flink-core/src/main/java/org/apache/flink/metrics/groups/ProxyMetricGroup.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.groups;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Metric group which forwards all registration calls to its parent metric group.
+ *
+ * @param <P> Type of the parent metric group
+ */
+@Internal
+public class ProxyMetricGroup<P extends MetricGroup> implements MetricGroup {
+	private final P parentMetricGroup;
+
+	public ProxyMetricGroup(P parentMetricGroup) {
+		this.parentMetricGroup = Preconditions.checkNotNull(parentMetricGroup);
+	}
+
+	@Override
+	public final void close() {
+		parentMetricGroup.close();
```

Comment:
It is safer to not do anything here. `close()` should only be called by the Task on the `TaskMetricGroup`, this would open us to the possibility of components closing the TaskMG as well.

There's also the looming StackOverflow when someone puts `ioMetrics.close()` into the `TaskMetricGroup#close()`.

Now that i think about it i believe `close()` (and by extension, `isClosed()`) has no business being in the MetricGroup interface in the first place, as users actually don't need to call it.